File: account_manager.py
import json
import os
from telethon import TelegramClient
from telethon.sessions import StringSession
from config import API_ID, API_HASH

class AccountManager:
    def __init__(self):
        self.accounts = {}
        self.load_accounts()
        
    def load_accounts(self):
        """Загружает данные аккаунтов из файла"""
        try:
            with open('accounts.json', 'r') as f:
                self.accounts = json.load(f)
        except FileNotFoundError:
            pass
            
    def save_accounts(self):
        """Сохраняет данные аккаунтов в файл"""
        with open('accounts.json', 'w') as f:
            json.dump(self.accounts, f)
            
    async def add_account(self, phone: str, proxy=None):
        """Добавляет новый аккаунт"""
        # Используем StringSession для сохранения сессии в строке
        client = TelegramClient(StringSession(), API_ID, API_HASH, proxy=proxy)
        await client.start(phone)
        
        # Сохраняем строку сессии
        session_string = client.session.save()
        
        self.accounts[phone] = {
            'session': session_string,
            'proxy': proxy
        }
        self.save_accounts()
        return client
    
    async def get_client(self, phone: str):
        """Получает клиент для аккаунта"""
        if phone not in self.accounts:
            return None
        
        account = self.accounts[phone]
        # Создаем клиент из сохраненной строки сессии
        client = TelegramClient(
            StringSession(account['session']),
            API_ID,
            API_HASH,
            proxy=account['proxy']
        )
        await client.start()
        return client

File: src/accountManager.js
import { TelegramClient } from 'telegram';
import { StringSession } from 'telegram/sessions/index.js';
import fs from 'fs';

export class AccountManager {
  constructor() {
    this.accounts = [];
    this.proxies = [];
    this.loadData();
  }

  loadData() {
    try {
      if (fs.existsSync('./data/accounts.json')) {
        this.accounts = JSON.parse(fs.readFileSync('./data/accounts.json', 'utf8'));
      }
      if (fs.existsSync('./data/proxies.json')) {
        this.proxies = JSON.parse(fs.readFileSync('./data/proxies.json', 'utf8'));
      }
    } catch (err) {
      console.error('Ошибка при загрузке данных:', err);
    }
  }

  saveData() {
    try {
      if (!fs.existsSync('./data')) {
        fs.mkdirSync('./data');
      }
      fs.writeFileSync('./data/accounts.json', JSON.stringify(this.accounts, null, 2));
      fs.writeFileSync('./data/proxies.json', JSON.stringify(this.proxies, null, 2));
    } catch (err) {
      console.error('Ошибка при сохранении данных:', err);
    }
  }

  async addAccount(phone, apiId, apiHash) {
    const session = new StringSession('');
    const client = new TelegramClient(session, apiId, apiHash, {
      connectionRetries: 5
    });

    await client.start({
      phoneNumber: phone,
      password: async () => await this.promptPassword(),
      phoneCode: async () => await this.promptCode(),
      onError: err => console.log(err),
    });

    const sessionString = client.session.save();
    this.accounts.push({ phone, apiId, apiHash, session: sessionString });
    this.saveData();
    
    return client;
  }

  async promptPassword() {
    const { password } = await inquirer.prompt([
      {
        type: 'password',
        name: 'password',
        message: 'Введите пароль двухфакторной аутентификации:'
      }
    ]);
    return password;
  }

  async promptCode() {
    const { code } = await inquirer.prompt([
      {
        type: 'input',
        name: 'code',
        message: 'Введите код подтверждения:'
      }
    ]);
    return code;
  }

  getAccounts() {
    return this.accounts;
  }

  addProxy(proxy) {
    this.proxies.push(proxy);
    this.saveData();
  }

  getProxies() {
    return this.proxies;
  }
}